### PR TITLE
Fixes #2945. Fix invalid_null_aware_operator in null-aware elements tests

### LINKED ISSUE #2945: Some of the tests for the Null-aware element language feature needs an update to reflect the Analyzer warnings
The following tests need to be updated to include the locations of the warnings soon to be reported by the Analyzer for the cases where a null-aware element or a map entry is of non-nullable static type.

## co19/LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_A05_t02

```unexpected analysis errors in syntax_A05_t02.dart:
- Line 32, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 39, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 46, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 47, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 52, column 19: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 57, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 57, column 21: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 58, column 23: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.
```

## co19/LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_A05_t04

```
unexpected analysis errors in syntax_A05_t04.dart:
- Line 65, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.
```

## co19/LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_parenthesized_A01_t01

```
unexpected analysis errors in syntax_parenthesized_A01_t01.dart:
- Line 46, column 12: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.
```

## co19/LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_unary_prefix_A01_t01

```
unexpected analysis errors in syntax_unary_prefix_A01_t01.dart:
- Line 48, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 49, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 50, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 51, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 56, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 57, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 58, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 59, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 64, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 65, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 66, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 67, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 68, column 8: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 69, column 8: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 70, column 8: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 71, column 8: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 76, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 76, column 12: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 77, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 77, column 12: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 78, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 78, column 12: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 79, column 5: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.

- Line 79, column 12: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.
```

## co19/LanguageFeatures/Control-flow-collections/Null-aware-elements/type_inference_map_A06_t01

```
unexpected analysis errors in type_inference_map_A06_t01.dart:
- Line 63, column 59: STATIC_WARNING.INVALID_NULL_AWARE_OPERATOR
  The receiver can't be null, so the null-aware operator '?' is unnecessary.
```

### PR DESCRIPTION
**Pull request by sgrekhov**
It can be argued but I choose the following approach for fixing this issue:
- `syntax_A05_t02.dart` and `syntax_A05_t04.dart` testing that parser can distingish null-aware element and if-null and null-aware expressions. So, simply ignore these warnings.
- `syntax_parenthesized_A01_t01.dart` typo fixed
- `syntax_unary_prefix_A01_t01.dart` it's syntax test again. But let's have two versions of the test here. One static and one run-time (which ignores the warnings). It'll check that warnings are really reported.
- `type_inference_map_A06_t01.dart` it's a type inference test. Expect a warning, no need the run-time version of the test.

Please review.

### CODE REVIEW

File: LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_A05_t02.dart [hunk 1]
```diff
@@ -29,33 +29,45 @@ int? f(int? v) => v;
...
+    ? f(null) ?? null: f(0) ?? null,
+
+    // 4: ?null
+    f(4) ?? null: ? f(null) ?? null,
+    // 5: ?5
+    f(5) ?? null: ? f(5) ?? null,
+    // 6: ?6
+    f(null) ?? 6: ? f(null) ?? 6,  // ignore: invalid_null_aware_operator
+    // null: ?7
+    f(null) ?? null: f(null) ?? 7,
```
Comment (eernstg) [inline LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_A05_t02.dart:61]: Perhaps?:
```suggestion
    f(null) ?? null: ? f(null) ?? 7,
```

File: LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_unary_prefix_A01_t01.dart [hunk 2]
```diff
@@ -46,37 +44,74 @@ main() {
 
   var list = [
     ?++e1,
+//  ^
+// [analyzer] unspecified
```
Comment (eernstg) [inline LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_unary_prefix_A01_t01.dart:48]: Isn't it a bug to report an error here? `e1 = e1 + 1` would have the type `int?`, which wouldn't (shouldn't) give rise to an `INVALID_NULL_AWARE_OPERATOR` warning.

**Review by eernstg — COMMENTED** [review]
Looks good! I noticed one typo, and then I'm questioning a bunch of errors around `?++i` and such (I can't see why we should get an error in those cases)

File: LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_A05_t02.dart (hunk 1 above)
Comment (sgrekhov) [reply LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_A05_t02.dart]: Of course! Thank yoou for noticing!

File: LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_unary_prefix_A01_t01.dart (hunk 2 above)
Comment (sgrekhov) [reply LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_unary_prefix_A01_t01.dart]: Reverted the change in this test and added a note to https://dart-review.googlesource.com/c/sdk/+/391621

**Review by sgrekhov — COMMENTED** [review]
Updated. PTAL.

File: LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_unary_prefix_A01_t01.dart (hunk 2 above)
Comment (eernstg) [reply LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_unary_prefix_A01_t01.dart]: OK, thanks!

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**